pkg/scheduler/framework/runtime/framework.go

```diff
@@ -1028,7 +1032,9 @@ func (f *frameworkImpl) WaitOnPermit(ctx context.Context, pod *v1.Pod) (status *
 		if s.IsUnschedulable() {
 			msg := fmt.Sprintf("pod %q rejected while waiting on permit: %v", pod.Name, s.Message())
 			klog.V(4).Infof(msg)
-			return framework.NewStatus(s.Code(), msg)
+			finalStatus := framework.NewStatus(s.Code(), msg)
```

Comment:
perhaps we should append the reasons we get from `s` instead of overriding it.

Reply:
I'm thinking that we can just log the message, and keep the reasons as is. Appending a reason like "reject pod ... by permit plugin" isn't that necessay. WDYT?

Reply:
sounds good